process/block/sovereignChainBlock.go

```diff
@@ -0,0 +1,1279 @@
+		err = process.ErrRootStateDoesNotMatch
+		return nil, nil, err
+	}
+
+	err = scbp.verifyValidatorStatisticsRootHash(headerHandler)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return headerHandler, newBody, nil
+}
+
+// checkExtendedShardHeadersValidity checks if used extended shard headers are valid as construction
+func (scbp *sovereignChainBlockProcessor) checkExtendedShardHeadersValidity() error {
+	lastCrossNotarizedHeader, _, err := scbp.blockTracker.GetLastCrossNotarizedHeader(core.SovereignChainShardId)
+	if err != nil {
+		return err
+	}
+
+	log.Trace("checkExtendedShardHeadersValidity", "lastCrossNotarizedHeader nonce", lastCrossNotarizedHeader.GetNonce())
+	extendedShardHdrs := scbp.sortExtendedShardHeadersForCurrentBlockByNonce()
+	if len(extendedShardHdrs) == 0 {
+		return nil
+	}
+
+	for _, extendedShardHdr := range extendedShardHdrs {
+		log.Trace("checkExtendedShardHeadersValidity", "extendedShardHeader nonce", extendedShardHdr.GetNonce())
+		err = scbp.headerValidator.IsHeaderConstructionValid(extendedShardHdr, lastCrossNotarizedHeader)
+		if err != nil {
+			return fmt.Errorf("%w : checkExtendedShardHeadersValidity -> isHdrConstructionValid", err)
+		}
+
+		lastCrossNotarizedHeader = extendedShardHdr
+	}
+
+	return nil
+}
+
+func (scbp *sovereignChainBlockProcessor) checkAndRequestIfExtendedShardHeadersMissing() {
+	orderedExtendedShardHeaders, _ := scbp.blockTracker.GetTrackedHeaders(core.SovereignChainShardId)
+
+	err := scbp.requestHeadersIfMissing(orderedExtendedShardHeaders, core.SovereignChainShardId)
+	if err != nil {
+		log.Debug("checkAndRequestIfExtendedShardHeadersMissing", "error", err.Error())
+	}
+}
+
+func (scbp *sovereignChainBlockProcessor) requestMissingHeaders(missingNonces []uint64, shardId uint32) {
+	for _, nonce := range missingNonces {
+		scbp.addHeaderIntoTrackerPool(nonce, shardId)
+		go scbp.extendedShardHeaderRequester.RequestExtendedShardHeaderByNonce(nonce)
+	}
+}
+
+func (scbp *sovereignChainBlockProcessor) sortExtendedShardHeadersForCurrentBlockByNonce() []data.HeaderHandler {
+	hdrsForCurrentBlock := make([]data.HeaderHandler, 0)
+
+	scbp.hdrsForCurrBlock.mutHdrsForBlock.RLock()
+	for _, headerInfo := range scbp.hdrsForCurrBlock.hdrHashAndInfo {
+		hdrsForCurrentBlock = append(hdrsForCurrentBlock, headerInfo.hdr)
+	}
+	scbp.hdrsForCurrBlock.mutHdrsForBlock.RUnlock()
+
+	process.SortHeadersByNonce(hdrsForCurrentBlock)
+
+	return hdrsForCurrentBlock
+}
+
+func (scbp *sovereignChainBlockProcessor) verifyCrossShardMiniBlockDstMe(sovereignChainHeader data.SovereignChainHeaderHandler) error {
+	miniBlockExtendedShardHeaderHashes, err := scbp.getAllMiniBlockDstMeFromExtendedShardHeaders(sovereignChainHeader)
+	if err != nil {
+		return err
+	}
+
+	crossMiniBlockHashes := sovereignChainHeader.GetMiniBlockHeadersWithDst(core.SovereignChainShardId)
+	for hash := range crossMiniBlockHashes {
+		if _, ok := miniBlockExtendedShardHeaderHashes[hash]; !ok {
```

Comment:
split in multiple lines?

Reply:
Ok, then we need to refactor the whole project to be this "go idiomatic"
Please split into 2 lines.